src/Lynx.purs

```diff
@@ -0,0 +1,277 @@
+module Lynx where
+
+import Prelude
+
+import Data.Const (Const)
+import Data.Either (Either)
+import Data.Either.Nested (type (\/))
+import Data.Foldable (fold, foldMap, for_)
+import Data.Functor.Coproduct.Nested (type (<\/>))
+import Data.Map (Map)
+import Data.Map as Data.Map
+import Data.Maybe (Maybe(..))
+import Data.Maybe as Data.Maybe
+import Effect.Aff.Class (class MonadAff)
+import Halogen as H
+import Halogen.Component.ChildPath (cp1, cp2, cp3)
+import Halogen.HTML as HH
+import Halogen.HTML.Events as HE
+import Halogen.HTML.Properties as HP
+import Lynx.Data.Expr (EvalError(..), Expr, ExprType, Key, boolean_, cents_, datetime_, print, reflectType, string_, toArray, toBoolean, toCents, toDateTime, toObject, toPair, toString)
+import Lynx.Data.Form (Field, Input(..), InputSource(..), Page, Section, ValidationError(..), asyncFromTypeahead, errorsToArray, getValue)
+import Lynx.Data.Form as Lynx.Data.Form
+import Network.RemoteData (RemoteData(..), fromEither)
+import Ocelot.Block.Button as Button
+import Ocelot.Block.Card as Card
+import Ocelot.Block.FormField as FormField
+import Ocelot.Block.Format as Format
+import Ocelot.Block.Input as Input
+import Ocelot.Block.ItemContainer (boldMatches)
+import Ocelot.Block.Layout as Layout
+import Ocelot.Block.Toggle (toggle) as Toggle
+import Ocelot.Component.DateTimePicker as DateTimePicker
+import Ocelot.Component.Dropdown as Dropdown
+import Ocelot.Component.Dropdown.Render as Dropdown.Render
+import Ocelot.Component.Typeahead as Typeahead
+import Ocelot.Data.Currency (parseCentsFromDollarStr)
+import Ocelot.HTML.Properties (css)
+
+type State =
+  { form :: RemoteData EvalError { expr :: Page Expr, evaled :: Page ExprType }
```

Comment:
I don't think we need `RemoteData` here anymore, right? That's now a concern of the parent. I think this can just be
```purescript
{ form :: Either EvalError { expr :: Page Expr, evaled :: Page ExprType }
...
}
```
Or do you think it makes sense to split these out now to something like
```purescript
{ expr :: Page Expr
, evaled :: Either EvalError (Page ExprType)
...
}

Reply:
Oh nice!

So what's the difference between those two? One is saying that you either have some error from trying to evaluate the page or you have both the page and its evaluated form. The other is saying that you always have the page and either some error from trying to evaluate the page or its evaluated form. Which one is the correct semantics?

Reply:
I think I'm going to drop the two cases of `RemoteData` for now, and we can think about changing the semantics later.

Reply:
> So what's the difference between those two? One is saying that you either have some error from trying to evaluate the page or you have both the page and its evaluated form. The other is saying that you always have the page and either some error from trying to evaluate the page or its evaluated form. Which one is the correct semantics?

 I think the latter, no? Because we do always have the page, just sometimes we'll run into errors when evaluating it. Plus it allows us to recover the unevaluated page upon failure, if we for some reason needed to.

Reply:
MAkes senes. I'll throw together a PR for it.